### PR TITLE
[Maintenance] Sylius 1.12 support

### PR DESCRIPTION
**Pull request by NoResponseMate**

### CODE REVIEW

File: tests/Application/Kernel.php [hunk 1]
```diff
@@ -58,13 +59,28 @@ protected function configureContainer(ContainerBuilder $container, LoaderInterfa
...
-        foreach ($this->getConfigurationDirectories() as $confDir) {
-            $this->loadRoutesConfiguration($routes, $confDir);
+        if (is_a($routes, RoutingConfigurator::class)) {
+            foreach ($this->getConfigurationDirectories() as $confDir) {
+                $this->loadRoutesConfiguration($routes, $confDir);
+            }
+        } else {
+            foreach ($this->getConfigurationDirectories() as $confDir) {
+                $this->loadRoutesCollection($routes, $confDir);
+            }
```
Comment (Rafikooo) [inline tests/Application/Kernel.php]:
```suggestion
        if (is_a($routes, RoutingConfigurator::class)) {
            foreach ($this->getConfigurationDirectories() as $confDir) {
                $this->loadRoutesConfiguration($routes, $confDir);
            }
            
            return;
        }
        
        foreach ($this->getConfigurationDirectories() as $confDir) {
            $this->loadRoutesCollection($routes, $confDir);
        }
```

**Review by Rafikooo — APPROVED** [review]

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -23,22 +23,22 @@ jobs:
...
-                php: [ "7.4", "8.0" ]
-                symfony: [ "^4.4", "5.4.*" ]
-                sylius: [ "1.10.*", "1.11.*" ]
-                node: [ "10.x" ]
-                mysql: [ "8.0" ]
-
+                php: ["8.0"]
+                symfony: ["^4.4", "^5.4"]
+                sylius: ["1.11.*", "1.12.*"]
+                node: ["12.x"]
```
Comment (GSadee) [inline .github/workflows/build.yml:29]: We should probably upgrade node version here to 16

File: composer.json [hunk 3]
```diff
@@ -23,11 +23,11 @@
         }
     ],
     "require": {
-        "php": "^7.4 || ^8.0",
+        "php": "^8.0",
         "friendsofsymfony/oauth-server-bundle": ">2.0.0-alpha.0 ^2.0@dev",
         "sylius-labs/doctrine-migrations-extra-bundle": "^0.1.3",
         "sylius-labs/polyfill-symfony-framework-bundle": "^1.0",
-        "sylius/sylius": "^1.10 || ^1.11",
+        "sylius/sylius": "~1.11.0 || ~1.12.0",
```
Comment (GSadee) [inline composer.json:30]: It would be nice to drop support for 1.10 in a separate PR

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Jan! :1st_place_medal:

**Pull request merged**